src/common/src/texpr.c

```diff
@@ -2041,13 +2061,20 @@ void vectorMathFunc(int16_t functionId, tExprOperandInfo *pInputs, int32_t numIn
     if (!hasNullInputs) {
       switch (functionId) {
         case TSDB_FUNC_SCALAR_LOG: {
-          assert(numInputs == 2);
-          double base = 0;
-          GET_TYPED_DATA(base, double, pInputs[1].type, inputData[1]);
+          assert(numInputs == 1 || numInputs == 2);
+          double base = M_E;
+          if (numInputs == 2) {
+            GET_TYPED_DATA(base, double, pInputs[1].type, inputData[1]);
+          }
           double v1 = 0;
           GET_TYPED_DATA(v1, double, pInputs[0].type, inputData[0]);
-          double result = log(v1) / log(base);
-          SET_TYPED_DATA(outputData, pOutput->type, result);
+          if (numInputs == 2) {
```

Comment:
if is not necessary here, because base has been initialized regardless of numInputs=1 or 2, here we just set result=log(v1)/log(base). In case of numInputs=1, log(base) is exactly 1, right?

Reply:
Improve precision. log(M_E) is not exactly 1.

Reply:
save two time-consuming computations, log(M_E) and double division when numInput is 1.

Reply:
ok, accepted